lib/oidc/dpop.ts

```diff
@@ -0,0 +1,240 @@
+
+      req.onerror = function () {
+        reject(req.error!);
+      };
+
+      req.onupgradeneeded = function () {
+        const db = req.result;
+        db.createObjectStore(DB_KEY);
+      };
+
+      req.onsuccess = function () {
+        const db = req.result;
+        const tx = db.transaction(DB_KEY, 'readwrite');
+
+        tx.onerror = function () {
+          reject(tx.error!);
+        };
+
+        const store = tx.objectStore(DB_KEY);
+
+        resolve(store);
+
+        tx.oncomplete = function () {
+          db.close();
+        };
+      };
+    }
+    catch (err) {
+      reject(err);
+    }
+  });
+}
+
+// convenience abstraction for wrapping IDBObjectStore methods in promises
+async function invokeStoreMethod (method: StoreMethod, ...args: any[]): Promise<IDBRequest> {
+  const store = await keyStore();
+  return new Promise((resolve, reject) => {
+    // https://github.com/microsoft/TypeScript/issues/49700
+    // https://github.com/microsoft/TypeScript/issues/49802
+    // @ts-expect-error ts(2556)
+    const req = store[method](...args);
+    req.onsuccess = function () {
+      resolve(req);
+    };
+    req.onerror = function () {
+      reject(req.error);
+    };
+  });
+}
+
+async function storeKeyPair (pairId: string, keyPair: CryptoKeyPair) {
+  await invokeStoreMethod('add', keyPair, pairId);
+  return keyPair;
+}
+
+// attempts to find keyPair stored at given key, otherwise throws
+export async function findKeyPair (pairId?: string): Promise<CryptoKeyPair> {
+  if (pairId) {
+    const req = await invokeStoreMethod('get', pairId);
+    if (req.result) {
+      return req.result;
+    }
+  }
+
+  // defaults to throwing unless keyPair is found
+  throw new AuthSdkError(`Unable to locate dpop key pair required for refresh${pairId ? ` (${pairId})` : ''}`);
+}
+
+export async function clearDPoPKeyPair (pairId: string): Promise<void> {
+  await invokeStoreMethod('delete', pairId);
+}
+
+export async function clearAllDPoPKeyPairs (): Promise<void> {
+  await invokeStoreMethod('clear');
+}
+
+// generates a crypto (non-extractable) private key pair and writes it to indexeddb, returns key (id)
+export async function createDPoPKeyPair (): Promise<{keyPair: CryptoKeyPair, keyPairId: string}> {
+  // TODO: clear exist keys when creating new one?
+  const keyPairId = cryptoRandomValue(4);
+  const keyPair = await generateKeyPair();
+  await storeKeyPair(keyPairId, keyPair);
+  return { keyPair, keyPairId };
+}
+
+// will clear PK from storage if certain token conditions are met
+/* eslint max-len: [2, 132], complexity: [2, 12] */
+export async function clearDPoPKeyPairAfterRevoke (revokedToken: 'access' | 'refresh', tokens: Tokens): Promise<void> {
+  let shouldClear = false;
+
+  const { accessToken, refreshToken } = tokens;
+
+  // revoking access token and refresh token doesn't exist
+  if (revokedToken === 'access' && accessToken && accessToken.tokenType === 'DPoP' && !refreshToken) {
```

Comment:
IMHO there's a potential naming confusion between token type 
https://github.com/okta/okta-auth-js/blob/4dc55e9e8591af45fdd0134d00f5b9e40b7ec521/lib/oidc/types/Token.ts#L25
and token kind
https://github.com/okta/okta-auth-js/blob/4dc55e9e8591af45fdd0134d00f5b9e40b7ec521/lib/oidc/types/Token.ts#L45
Maybe add comment to line 25 about possible values (`Bearer` or `DPoP`)?